### PR TITLE
Rate limiting

### PR DESCRIPTION
**Pull request by abdelrahman-mos**
added rate limiting to all routes

### CODE REVIEW

File: src/middlewares/rate-limiter.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import rateLimiter from "express-rate-limit";
+
+export const limiter = rateLimiter({
+  max: 120,
```
Comment (Deadreyo) [inline src/middlewares/rate-limiter.ts:4]: The limit of 120 per 5 seconds is too big and doesn't help. Limit it to 20 requests per 60 seconds.

File: src/routers/auth.router.ts [hunk 2]
```diff
@@ -1,32 +1,47 @@
...
+				}
+			}
+		}*/
+    next();
+  },
+  mwValidateLoginData,
+  Login
 );
 
+authRouter.use(limiter);
```
Comment (Deadreyo) [inline src/routers/auth.router.ts:32]: You used the limiter on global router already, no need to have it here.

File: src/routers/participant.router.ts [hunk 3]
```diff
@@ -1,60 +1,50 @@
...
-)
-.post("/add", mwValidateParticipant, addParticipant)
-.patch(
-    "/update",
-    isLoggedIn,
-    giveAccessTo("hr"),
-    updateParticipantByPhone
-)
-.delete(
+  .use(limiter)
```
Comment (Deadreyo) [inline src/routers/participant.router.ts:26]: You used the limiter on global router already, no need to have it here

File: src/routers/participant.router.ts [hunk 4]
```diff
@@ -1,60 +1,50 @@
...
+  .get("/getAll", isLoggedIn, giveAccessTo("member"), getAllParticipants)
+  .post("/add", mwValidateParticipant, addParticipant)
```
Comment (Deadreyo) [inline src/routers/participant.router.ts:28]: The `/add` endpoint in participants need a dedicated limiter to limit it stricter than the others. Could be limited to 4 submissions per 2 hours for example.

**Review by nightknighto — CHANGES_REQUESTED** [review]
Just some small changes needed

**Review by nightknighto — APPROVED** [review]
Done the requested changes myself

**Pull request merged**